libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -206,7 +206,7 @@ constructor(
             )
         }
 
-        fasterRouteController = FasterRouteController(directionsSession, tripSession, logger)
+        fasterRouteController = FasterRouteController(directionsSession, tripSession, RouteRefreshAdapterProvider.provideRouteRefreshAdapter(), logger)
```

Comment:
Confused why route refresh would be used for faster route, and not route refresh 🤔 

Also, this will give a different RouteRefreshAdapter than the one given in the NavigationOptions

Reply:
yes, this is intention. User might overwrite `reroute` logic, but not `faster route`

Reply:
anyway, I've made changes a bit and added docs, see `RoutingOptionsProvider` 👀

Reply:
ok i think it's the "RouteRefresh" in RouteRefreshAdapter that was confusing for me.

am liking the idea of a RoutingOptionsProvider though!